app/models/dataset_record/darwin_core/occurrence.rb

```diff
@@ -540,15 +540,48 @@ def parse_record_level_class
     # institutionCode: [repository.acronym] # TODO: Use mappings like with namespaces here as well? (Although probably attempt guessing)
     institution_code = get_field_value(:institutionCode)
     if institution_code
-      repository = Repository.find_by(acronym: institution_code)
+      repository = nil
+      error_messages = []
+      acronym_repositories = Repository.where(acronym: institution_code)
+      if acronym_repositories.count == 1
+        repository = acronym_repositories.first
+      elsif acronym_repositories.count > 1
+        error_messages << "Multiple repositories with acronym #{institution_code} found."
+      else
+        error_messages << "No repositories with acronym #{institution_code} found."
+      end
+
+      if !repository && (institution_code.starts_with?('http://') || institution_code.starts_with?('https://'))
```

Comment:
Would perhaps be best to test URL first, then acronym so you avoid hitting the database twice (first with potential URL string than cannot possibly be an acronym)?

Reply:
Agreed.